desktop-widgets/templatelayout.cpp

```diff
@@ -152,7 +152,12 @@ QString TemplateLayout::generateStatistics()
 
 QString TemplateLayout::readTemplate(QString template_name)
 {
-	QFile qfile(getPrintingTemplatePathUser() + QDir::separator() + template_name);
+	QString filename;
+	if (template_name.contains("template.html"))
+		filename = template_name;
+	else
+		filename = getPrintingTemplatePathUser() + QDir::separator() + template_name;
+	QFile qfile(filename);
```

Comment:
could you explain why was this change needed?

Reply:
I changed the semantics of the "Apply" button to: Update the preview to the current version of the template in the editor. (before Apply forced a "save"). So, unless "OK" is clicked, it should use a temporary template (the one form the editor window) which is not stored in the template directory but in the temporary directory of the printing job. That comes with an absolute file name.

Reply:
this seems fine, having a comment above might be easier for the readers to parse.
unless i'm mistaken it can also be `template_name == "template.html"`

Reply:
I added a comment and used a less generic 'magic' filename. I had thought about detecting a directory separator as the actual templates are just simple filenames while the temporary one has an absolute path but wanted to avoid a possible / vs \ confusion (afaik Qt even on windows can use / but I don't know if it actively does the conversion, i.e. if the automatically generated file has a / or a \).

Reply:
Widows should be fine with / but qt recommend using QDir::separator()

Reply:
My understanding is that Qt accepts both / and \, but we would need the other direction: are we guaranteed that an absolute path returned by QFile::fileName() will contain QDir::separator() and not possibly another character instead? That was my worry.

Reply:
>  are we guaranteed that an absolute path returned by QFile::fileName() will contain QDir::separator() and not possibly another character instead?

it would return the right separator for the OS.